applications/nrf5340_audio/src/audio/sw_codec_select.c

```diff
@@ -206,109 +235,89 @@ int sw_codec_decode(uint8_t const *const encoded_data, size_t encoded_size, bool
 	switch (m_config.sw_codec) {
 	case SW_CODEC_LC3: {
 #if (CONFIG_SW_CODEC_LC3)
-		char *pcm_in_data_ptrs[m_config.decoder.channel_mode];
+		if (bad_frame && IS_ENABLED(CONFIG_SW_CODEC_OVERRIDE_PLC)) {
+			/* Could just clear the stereo buffer and update the size to not have to do
+			 * resampling on a zero buffer.
+			 */
 
-		switch (m_config.decoder.channel_mode) {
-		case SW_CODEC_MONO: {
-			if (bad_frame && IS_ENABLED(CONFIG_SW_CODEC_OVERRIDE_PLC)) {
-				memset(decoded_data_mono[AUDIO_CH_L], 0, PCM_NUM_BYTES_MONO);
-				decoded_data_size = PCM_NUM_BYTES_MONO;
-			} else {
-				ret = sw_codec_lc3_dec_run(
-					encoded_data, encoded_size, LC3_PCM_NUM_BYTES_MONO, 0,
-					decoded_data_mono[AUDIO_CH_L],
-					(uint16_t *)&decoded_data_size, bad_frame);
-				if (ret) {
-					return ret;
-				}
-
-				ret = sw_codec_sample_rate_convert(
-					&decoder_converters[AUDIO_CH_L],
-					m_config.decoder.sample_rate_hz,
-					CONFIG_AUDIO_SAMPLE_RATE_HZ, decoded_data_mono[AUDIO_CH_L],
-					decoded_data_size,
-					decoded_data_mono_system_sample_rate[AUDIO_CH_L],
-					&pcm_in_data_ptrs[AUDIO_CH_L], &pcm_size_mono);
-				if (ret) {
-					LOG_ERR("Sample rate conversion failed for mono: %d", ret);
-					return ret;
-				}
-			}
+			memset(dec_data_stereo, 0, PCM_NUM_BYTES_MONO * m_config.decoder.num_ch);
+			decoded_data_size = PCM_NUM_BYTES_MONO * m_config.decoder.num_ch;
+		} else {
+			struct audio_data audio_data_tx;
+			struct audio_data audio_data_rx;
 
-			/* For now, i2s is only stereo, so in order to send
-			 * just one channel, we need to insert 0 for the
-			 * other channel
+			/* The setting of the meta data will happen when the audio data is passed
```

Comment:
Yes once the meta data is defined it can be set once, at this point it would be copied into the object at source. Could make the meta data a pointer and then have a static config that it points to. This would be updated after  the codec negotiation and removes the copy ect.